### PR TITLE
Add methods to control the state of drawers

### PR DESCRIPTION
**Pull request by pcomans**
This PR adds the following:
- Methods for checking if the drawer or end drawer are open
- Methods for closing them programmatically

All credit goes to @cforkish

### CODE REVIEW

**Comment by cforkish** [comment]
I confirm that I'm okay with my commits being contributed to this project.

File: packages/flutter/lib/src/material/drawer.dart [hunk 1]
```diff
@@ -287,6 +287,8 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
     }
   }
 
+  bool get isOpen => !_controller.isDismissed;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/drawer.dart]: this needs documentation

**Comment by Hixie** [comment]
Can you elaborate on the use cases here?

**Comment by Hixie** [comment]
This will need docs and tests, but first we should discuss the use case, especially for `isOpen`. For example, should widgets be automatically rebuilt when the value of `isOpen` changes?

**Comment by Hixie** [comment]
> We ended up doing this

Can you elaborate on this? When would you run this code?

> Can you elaborate on the following question?

I think you answered it to my satisfaction. I was just wondering if we needed a subscription mechanism (e.g. an InheritedWidget to expose the value), or if this was something that would only be used during event handlers or some such.

**Comment by pcomans** [comment]
@Hixie we use it to wrap the main screen of our app.
If the drawer is closed, the back button opens it. If it's already open, the back button exits the app.

```
    return new WillPopScope(
      child: new Scaffold(
        key: _globalKey,
        body: _mainScreen(),
      ),
      onWillPop: () {
        ScaffoldState scaffold = _globalKey.currentState;
        if (Platform.isAndroid && scaffold != null) {
          if (scaffold.isDrawerOpen) {
            SystemNavigator.pop();
          } else {
            scaffold.openDrawer();
          }
        }
        return new Future<bool>.value(false);
      },
    );
```

**Comment by pcomans** [comment]
I had to close this PR and created a new one at https://github.com/flutter/flutter/pull/14673.
I appologize for the inconvenience.

**Comment by goderbauer** [comment]
@jslavitz Can we close this now that your PR is submitted?

**Pull request closed without merging**